pkg/upgrade/hub.go

```diff
@@ -120,3 +130,48 @@ func (b *FListInfo) Version() (semver.Version, error) {
 
 	return b.extractVersion(b.Name)
 }
+
+// Absolute returns the actual flist name
+func (b *FListInfo) Absolute() string {
+	name := b.Name
+	if b.Type == "symlink" {
+		name = b.Target
+	}
+
+	return filepath.Join(b.Repository, name)
+}
+
+// Files gets the list of the files of an flist
+func (b *FListInfo) Files() ([]File, error) {
+	flist := b.Absolute()
+	if len(flist) == 0 {
+		return nil, fmt.Errorf("invalid flist info")
+	}
+
+	var content struct {
+		Content []File `json:"content"`
+	}
+
+	u, err := url.Parse(HubBaseURL)
+	if err != nil {
+		panic("invalid base url")
+	}
+
+	u.Path = filepath.Join("api", "flist", flist)
+	response, err := http.Get(u.String())
+	if err != nil {
+		return nil, err
+	}
+
+	defer response.Body.Close()
+	defer ioutil.ReadAll(response.Body)
```

Comment:
Whis is this needed ? you already consume the body with the json decoder

Reply:
the line below, I check on the status of the response, and then return an error if the status is not OK. in that case, the body will not be consumed, and hence the connection will not be reused (according to the docs)

So I found that it's a good practice to always consume the body in a `defer` to prevent this.